lib/config/admin.ts

```diff
@@ -1,30 +1,34 @@
 import type { RenovateConfig, RepoAdminConfig } from './types';
 
-let adminConfig: RepoAdminConfig = {};
+let adminConfig: RepoAdminConfig;
 
 // TODO: once admin config work is complete, add a test to make sure this list includes all options with admin=true (#9603)
-export const repoAdminOptions = [
-  'allowCustomCrateRegistries',
-  'allowPostUpgradeCommandTemplating',
-  'allowScripts',
-  'allowedPostUpgradeCommands',
-  'customEnvVariables',
-  'dockerChildPrefix',
-  'dockerImagePrefix',
-  'dockerUser',
-  'dryRun',
-  'exposeAllEnv',
-  'privateKey',
-];
+const adminDefaults: RepoAdminConfig = {
+  allowCustomCrateRegistries: undefined,
+  allowPostUpgradeCommandTemplating: undefined,
+  allowScripts: undefined,
+  allowedPostUpgradeCommands: undefined,
+  customEnvVariables: undefined,
+  dockerChildPrefix: undefined,
+  dockerImagePrefix: undefined,
+  dockerUser: undefined,
+  dryRun: undefined,
+  exposeAllEnv: undefined,
+  privateKey: undefined,
+  localDir: '',
+  cacheDir: '',
+};
 
 export function setAdminConfig(config: RenovateConfig = {}): void {
-  adminConfig = {};
-  for (const option of repoAdminOptions) {
-    adminConfig[option] = config[option];
-    delete config[option]; // eslint-disable-line no-param-reassign
+  adminConfig = { ...adminDefaults };
+  for (const [key, defaultValue] of Object.entries(adminDefaults)) {
+    adminConfig[key] = config[key] ?? defaultValue;
+    delete config[key]; // eslint-disable-line no-param-reassign
   }
 }
 
+setAdminConfig();
```

Comment:
This is not directly related to this work, but I am not a big fan of global mutable state. A singleton pattern could work here: [simplified example](https://www.typescriptlang.org/play?#code/MYGwhgzhAEBKCmAHA9gQQCYFsCWA7AwsrgGbYDm0A3gLABQ00iATtgG5gAu80EHn2waHl5hcweAC44SNFjyES5OnQbM2nbsCKkyUhFqboAPLxa4yAGmiiAngD4A3MvqMW7LtC25TAV2AdkJgAKAEoqFQZoAHoo6ABleA5odHhiMB8QDhgAC3gmeAiGDmzsCAA6Lx1oAF5wl0jrKQBGQugAXycXNudVHwAjEAEePg4hskSASW8+MXhQvRkMHAJtcjqG7GJoIIBCBBQl+VWyMuEZ8TCaesj92WWFHVPp0XEa6Fx4AHdpA7kVxTIoU6DW6tFa+Q4PiYuB+dyOAKeIlmwPaPUY-UGgnGHEO-x082kBmMpjwlmsuHs60iEKhMOKpQqxxRoIiiAxQwgiVxD3IQUq5AWRJMHDMZNsdkurXp5X5FFqsuZdBZtC8vE8xyab1u3OOZWxUyR4iBdFlTTKnJxfx5gMoYGabRCzlVSVlACYtYsrbr9c9ZsaVUQ1WAvQD3fLjq69VyQ-iQp0YlRoAAiMBJ5qogPeZAgeBlEDIQHB+4Rx1gzM6SMWnUAoKUPpSV0Ok2BpJF+E6TXhgFm7HV2Px2KUZN9NPQd2g1XZ3P5wsx8hNON0IA)

Reply:
👍 Maybe, we can do some generic class `Singleton<T>` for serving other cases like this

Reply:
https://github.com/renovatebot/renovate/issues/9996